pisa/stages/discr_sys/csv_hypersurfaces.py

```diff
@@ -0,0 +1,277 @@
+        else:
+            raise ValueError(
+                f"Unsupported type {type(nominal_systematics)} for nominal_systematics."
+            )
+        
+        self.inter_param = inter_param
+        self.propagate_uncertainty = propagate_uncertainty
+
+        expected_container_keys = [
+            'weights',
+        ]
+        if 'error_method' in std_kwargs:
+            if std_kwargs['error_method']:
+                expected_container_keys.append('errors')
+
+        # -- Initialize base class -- #
+        super().__init__(
+            expected_params=list(self.nominal_systematics.keys()) + [self.inter_param],
+            expected_container_keys=expected_container_keys,
+            **std_kwargs,
+        )
+        
+        if links is None:
+            self.links = {}
+        elif not isinstance(links, Mapping):
+            self.links = ast.literal_eval(links)
+        else:
+            self.links = links
+
+    # pylint: disable=line-too-long
+    def setup_function(self):
+        """Load the fit results from the file and make some check compatibility"""
+
+        self.data.representation = self.calc_mode
+        
+        self.hs = {}
+        for f in self.fit_results_file:
+            k = f.split('/')[-1].split('.')[0]
+            if k.startswith('hs_'):
+                k = k[3:]
+            self.hs[k] = pd.read_csv(find_resource(f))
+
+        if self.links is not None:
+            for key, val in self.links.items():
+                self.data.link_containers(key, val)
+
+        # create containers for scale factors
+        for container in self.data:
+            container["hs_scales"] = np.empty(container.size, dtype=FTYPE)
+            if self.propagate_uncertainty:
+                hs = self.hs[container.name]
+
+                # get uncertainty at nominal value of interpolation param
+                inter_param_value = self.params[self.inter_param].m
+                start_idx = np.argmin(np.abs(hs[self.inter_param] - inter_param_value))
+                _, c = np.unique(hs[self.inter_param], return_counts=True)
+                stop_idx = start_idx + c[0]
+
+                hs_scales_uncertainty = hs['intercept_sigma'][start_idx:stop_idx]
+                container["hs_scales_uncertainty"] = np.array(hs_scales_uncertainty).reshape(container.size)
+
+        # Check map names match between data container and hypersurfaces
+        for container in self.data:
+            assert container.name in self.hs, f"No match for map {container.name} found in the hypersurfaces"
+
+        self.data.unlink_containers()
+        
+    def get_corr_factors(self, hs, param_values):
+        """Get hypersurface correction factors."""
+        # Get difference between param_values and nominal.
+        diff = {k: v - self.nominal_systematics[k] for k, v in param_values.items()}
+
+        # Return correction factor.
+        return hs["intercept"] + sum([hs[k] * v for k, v in diff.items()])
```

Comment:
This line would read better if you replaced `k` - > `p` and `v` -> `d` (or `delta`) IMO.